### DISCUSSION TITLE
Next.js discussions!!

### DISCUSSION THREAD

**Discussion by ijjk** [vercel/next.js#10116] [Help] [answered]
Excited to see these being used

**Comment by mxstbr** [comment]
:wave:

**Comment by Timer** [comment]
Testing multiple answers

**Reply by becca** [reply]
we currently only support one answer per discussion.

**Reply by tunnckoCore** [reply]
did we? :D

**Reply by foxundermoon** [reply]
nice

**Reply by oldtimerza** [reply]
Is it possible to split discussions, so that if multiple answers do appear, then you can move answers into more specific discussions?

**Reply by becca** [reply]
not at the moment! is this something you have seen occur and can point me to (either on github or off)?

**Comment by martinbavio** [comment]
🤯

**Comment by ztanner** [comment]
Neat!

**Comment by coetry** [comment]
:open_mouth: :rocket:

**Comment by vinayakkulkarni** [comment]
What is this 🧙🏻‍♂️

**Comment by paulogdm** [comment]
Yay!
![giphy](https://user-images.githubusercontent.com/7690649/75060094-fa692100-54bc-11ea-892f-393be7569dae.gif)

**Comment by jescalan** [comment]
Hi friends

**Comment by ztanner** [comment]
oh shnap, reactions!

**Comment by martinbavio** [comment, marked as answer]
<img src="https://media1.tenor.com/images/2576fdc39e58b7f14c748daf9b3a4365/tenor.gif?itemid=7809153" />

**Comment by ztanner** [comment]
^ I think this should be the new accepted answer.

**Comment by f3l1x** [comment]
Wheeee!  🎉

**Comment by kachkaev** [comment]
GitHub notifications after Next.js enabled discussions 😅

![a lot of notifications.gif](https://media.tenor.com/images/5e791af3d94a60f185d77a3d4ca2f31a/tenor.gif)

**Comment by cyberomin** [comment]
Is this thing working?

**Comment by krismuniz** [comment]
I love this!

**Comment by JohnMilazzo** [comment]
We should discuss this!   😉

**Comment by TheSisb** [comment]
Hello World!

**Comment by emplums** [comment]
💖

**Comment by mknepprath** [comment]
Testing commenting in this discussion and hanging out with Cool Friends

**Comment by brianlovin** [comment]
I love me some Next.js!

**Comment by siddharthkp** [comment]
love what GitHub is doing 💛

**Comment by in-in** [comment]
> #10116  

But still, the counter is linked to issues
:thinking:

**Comment by nerual** [comment]
🌮 🦄

**Comment by ragingwind** [comment]
🏄‍♂️🌊

**Comment by LeoNatan** [comment]
> 
> 
> > #10116
> 
> But still, the counter is linked to issues
> 🤔

That's to allow linking to discussions in release notes, commits and elsewhere, without having a dedicated syntax.

**Comment by vinayakkulkarni** [comment]
Found a bug in notifications beta, even after I visit this page, the notification doesn't go away... 

Hmmm 🤔

**Comment by aysark** [comment]
Cool!

**Comment by timdorr** [comment]
How do I turn this on for my own repos?

**Comment by yowmamasita** [comment]
Pet peeve:

From [https://github.com/zeit/next.js/issues](https://github.com/zeit/next.js/issues) to /issues/123

[https://github.com/zeit/next.js/discussions](https://github.com/zeit/next.js/discussions) to /discussions/123

But...

[https://github.com/zeit/next.js/pulls](https://github.com/zeit/next.js/pulls) to /pull/123

Also:

`https://github.com/zeit/next.js/issues/1` redirects to /pull/1

`https://github.com/zeit/next.js/pull/16` redirects to /issues/16

But...

[https://github.com/zeit/next.js/discussions/1](https://github.com/zeit/next.js/discussions/1) and [https://github.com/zeit/next.js/discussions/16](https://github.com/zeit/next.js/discussions/16) are 404

**Comment by vinayakkulkarni** [comment]
> How do I turn this on for my own repos?

It's in beta.

**Comment by zachzhao1984** [comment]
Finally~

**Comment by eduarmreyes** [comment]
Test

**Comment by ouuan** [comment]
Shouldn't this be "Questions" instead of "Discussions"?

**Comment by ouuan** [comment]
> Definitely not. Questions is too specific, people may want to discuss things that are not necessarily questions.

I think the UI design is for questions (like "answered", "unanswered", "FAQ"). Can you give examples that are not questions but are better to discuss in "Discussions" rather than issues?

**Comment by zhe** [comment]
This is very useful for internal reference to certain issue or pull request. 🚀

**Comment by sebilasse** [comment]
> Is this thing working?

![comp_no](https://user-images.githubusercontent.com/750300/75088032-de5e9180-5547-11ea-8895-6f0bb9d6c774.jpeg)

**Comment by bruno-ds** [comment]
Honestly, I think stack overflow has gotten me: all I gonna say is that I want a clone of it, and I'm so disappointed about myself for that!

The answers (both the accepted one, but also the others) are hidden in the flood of debate. 
Shouldn't the debate be either grouped above the question or one answer, or a filter let you see only the answers (but then how could you read the debate that can enhance the quality of the answer)

**Comment by sebilasse** [comment]
@bruno-ds (not sure if we can notify, autofill doesn't gimme prev. speakers)

1) stack overflow has gotten me too
2) Fuck clones
3) Agreed : 
A clear visual indication for accepted answers
A better one for the others, also how about letting you reply to a particular answer …

What do I want (apart from github not being MS) : 

- Federated discussions via the world’s leading federated social web standard "ActivityPub"
- Get away from the strictly linear one layer thread, I want to visually clear reply to e.g. quotes

I mean:
This is a clone of "Issues".
Apart from adding another like count and putting a supersexy fully fledged proactive-green checkmark icon somewhere :

**Did anybody at github discuss how "Discussions" differ from "Issues" ???** 
Now I would like to see how these developers "discuss" in real life …

**Reply by becca** [reply]
👋 threads now available here! 🙇

**Reply by asmeurer** [reply]
I'm sorry, but I feel like threads are going to be a nightmare. I can't understand why people like them on anything TBH. Right now, when I go to a GitHub issue, I can scroll to the bottom and easily see what new comments have been made. With threads, I either have to scan the whole page, or else I am forced to interact via my notifications; I can't just go to an discussion page and look for what is new. This problem already sort of exists with PRs and line comments. Perhaps something that would help (both for discussions and PR line comments) would be if any new comment automatically pushed that comment to the bottom of the screen. That's harder with discussions because now the top-level comments are out of order, but I don't see any reason why that isn't done for PR line comments.

**Reply by becca** [reply]
@asmeurer thanks for this feedback. is the concern that triaging notifications for threads isn't as straightforward because they will not always be in a chronological list like issue comments are now?

**Reply by asmeurer** [reply]
I believe the notifications would not be an issue. The problem would be finding unread discussion replies after visiting the page. For instance, if I have a discussion page open in my browser and I come back to it after a day, how do I easily tell what is new. If it is a large discussion like this one, it isn't very easy, because you have to look through every reply. Contrast that to an issue where you can see what is new just be seeing which comments were added at the bottom of the page.

**Reply by becca** [reply]
ah ok. this is super helpful feedback, thank you Aaron. i'll see what we can do to make that experience easier especially for long posts.

**Comment by praveenpuglia** [comment]
and how is this different? Genuine question.

**Comment by outloudvi** [comment]
GitHub has Kanban
GitHub has CI/CD
GitHub has StackOverflow
...
What's next? GitHub Cloud Service?

**Comment by CAFxX** [comment]
> What's next? GitHub Cloud Service?

It:s called Azure

**Comment by sebilasse** [comment]
@vinayakkulkarni  Did you click "Unsubscribe" on the top ?

et al. Much luck in scrolling or browser-searching the comment I referred too ;)

**Comment by smakosh** [comment]
Hi 👋

**Comment by minazukie** [comment]
Hello~~👋

**Comment by olpeh** [comment]
Let's go!

**Comment by perminder-klair** [comment]
yo!

**Comment by fiatjaf** [comment]
Looks like an issue to me.

**Comment by menduo** [comment]
so cool.

**Comment by jsjoeio** [comment]
Could this be used in lieu of Slack and Discord for open source communities?

**Comment by brawaru** [comment]
@jsjoeio aren't they used for a little different purposes, though? Discussions on GitHub are for single topic threads (like questions or actual discussions), when Discord/Slack communities for… *chatting* about project as a whole with predefined number of topics (‘channels’).

**Comment by jsjoeio** [comment]
Yeah, I guess you're right @Sasha-Sorokin.

Since it's single topic, its probably more similar to Stack Overflow or Reddit (minus key features like upvoting).

I think I'm over Slack/Discord and was hoping for something new 😅

**Comment by srikwit** [comment]
Nested comment section like HN would be really cool!

**Reply by becca** [reply]
we just enabled nested comments for `/next.js` just now! ✨

**Comment by ahmadawais** [comment]
An extended version of issues or more features that I missed 🙃

**Comment by agustif** [comment]
At least would be better than spectrum.chat nextjs community. The app is buggy as hell for me on desktop/website.

**Reply by pooriamo** [reply]
spectrum.chat is terrible. It's more like a school project of a child.

**Comment by iagobruno** [comment]
👀

**Comment by vinayakkulkarni** [comment]
@jsjoeio : Look for: <img width="33" alt="Screenshot 2020-02-23 at 1 39 12 AM" src="https://user-images.githubusercontent.com/19776877/75098519-4d39fa00-55dd-11ea-9263-c08d9de3b5e6.png">

Tiny icon on the right, you can click on it to upvote the comment.

**Comment by jsjoeio** [comment]
@vinayakkulkarni thanks for pointing that out! I didn't even realize 😄

**Comment by shtakai** [comment]
:sushi:

**Comment by AmericanWebbery** [comment]
What did I just walk into?

**Comment by jeremyBanks** [comment]
Apple Discussions meets Stack Overflow

**Reply by jeremyBanks** [reply]
Nice

**Reply by jeremyBanks** [reply]
https://github.com/zeit/next.js/discussions/10116#discussioncomment-569

**Reply by bruno-ds** [reply]
test thread

**Comment by dhsathiya** [comment]
This is super :cool:!

**Comment by asmeurer** [comment]
What does the upvote feature do? It doesn't seem to do anything other than incrementing a number. Will it re-order comments?

And more importantly, will there be a way to disable it? I'm not particularly keen on GitHub becoming like Reddit or Hacker News.

**Reply by becca** [reply]
we're about to 🚢 a feature to reorder comments in posts. it will reorder so the answer comes first (after the initial post) and then order by votes. should be available shortly.

**Comment by tunnckoCore** [comment]
It's interesting that there's nowhere any announcement or blog post or anything about Github Discussions. There's only a link in hackernews linking to this.

@becca, it's "private beta" or what?

Anyway, it looks great :)

**Reply by becca** [reply]
Correct. We're currently experimenting with ZEIT and welcome your feedback so we can continue iterating and expanding Discussions to other communities.

**Reply by erlend-sh** [reply]
Is there a process for other orgs to apply to be beta testers?

**Reply by becca** [reply]
@erlend-sh you can use the feedback link in the right hand sidebar next to the `beta` tag - i triage those requests 👋

**Comment by aralroca** [comment]
Yeah! This is going to replace spectrum?!!

**Reply by Timer** [reply]
Yes

**Comment by foxundermoon** [comment]
how to enable on my repository?

**Reply by becca** [reply]
Discussions is currently limited as an experiment to ZEIT. Please continue sharing your feedback as we get closer to enabling this within other communities 😄

**Reply by natedunn** [reply]
> Discussions is currently limited as an experiment to ZEIT. Please continue sharing your feedback as we get closer to enabling this within other communities 😄

You got it 👍

**Comment by shabbir-ahmed** [comment]
@ijjk 

Apologize for this question! Is there any example of Next.js 9 with Socket.io? I have found one [article](https://www.valentinog.com/blog/socket-react/) but it is with a custom server but I want without a custom server like [this](https://nextjs.org/docs/api-routes/introduction)

It will be really appreciated if you help.

Thanks
